src/jit/lower.cpp

```diff
@@ -809,6 +808,40 @@ GenTreePtr Lowering::NewPutArg(GenTreeCall* call, GenTreePtr arg, fgArgTabEntryP
     isOnStack = info->regNum == REG_STK;
 #endif // !FEATURE_UNIX_AMD64_STRUCT_PASSING
 
+#ifdef _TARGET_ARM_
+    // Strcut can be split into register(s) and stack on ARM
+    if (info->isSplit)
+    {
+        if (arg->OperGet() != GT_OBJ)
+        {
+            NYI_ARM("Lowering: Oper for struct argument is not GT_OBJ");
+        }
+
+        putArg = new (comp, GT_PUTARG_SPLIT)
+            GenTreePutArgSplit(arg, info->slotNum PUT_STRUCT_ARG_STK_ONLY_ARG(info->numSlots), info->numRegs,
+                               info->isHfaRegArg, call->IsFastTailCall(), call);
+        if (arg->gtFlags & GTF_LATE_ARG)
+        {
+            putArg->gtFlags |= GTF_LATE_ARG;
+        }
+        else
+        {
+            info->node = putArg;
+        }
+
+        // Set GC Pointer info
+        BYTE*    gcLayout = new (comp, CMK_Codegen) BYTE[info->numSlots + info->numRegs];
+        unsigned numRefs  = comp->info.compCompHnd->getClassGClayout(arg->gtObj.gtClass, gcLayout);
+        putArg->AsPutArgSplit()->setGcPointers(numRefs, gcLayout);
+
+        JITDUMP("new node is : ");
+        DISPNODE(putArg);
+        JITDUMP("\n");
+
+        return putArg;
+    }
+#endif
+
```

Comment:
Couldn't the end of this be:
```
    else
#endif // _TARGET_ARM_
```

in which case, you wouldn't need to include this in your `if`:
```
        if (arg->gtFlags & GTF_LATE_ARG)
        {
            putArg->gtFlags |= GTF_LATE_ARG;
        }
        else
        {
            info->node = putArg;
        }

...

        JITDUMP("new node is : ");
        DISPNODE(putArg);
        JITDUMP("\n");

        return putArg;
    }
#endif
```
?

(it looks like `updateArgTable` is always `true` so you could delete it.)